protobuf/CHANGELOG.md

```diff
@@ -1,3 +1,17 @@
+## Unreleased
```

Comment:
I think the typical pattern we use in other repos is to start a `-dev` version whenever you make a change by updating `pubspec.yaml` and `CHANGELOG.md`, e.g. in this case `protobuf` package is currently `2.0.2-dev`. You can update `pubspec.yaml` to use this version and then replace `Unreleased` with `2.0.2-dev`.

See https://github.com/dart-lang/sdk/wiki/External-Package-Maintenance#making-a-change

At some point I thought about creating a CI check to enforce this (e.g. if you have PR with changes to `lib/` you need to have version set to `-dev`). - but I did not get time.

Reply:
> But it's still technically a breaking change

I think we should put the reserved names back even though we don't need to reserve them, to keep things backwards compatible. Then add a note and open an issue to remove then in the next major release. I will do that in a separate PR.

Reply:
Done in #636.

Reply:
- Reverted the breaking change
- Removed changelog entry for the breaking change
- Bumped package verson to 2.0.2-dev